### PR TITLE
adding new basic tests

### PR DESCRIPTION
**Pull request by wsenafranca**
added new tests for beta, exponential, lognormal and normal distributions from #1820

### CODE REVIEW

File: packages/base/tests/functional/basics/Random.lua [hunk 1]
```diff
@@ -24,6 +24,148 @@
...
+		Random{seed = 10}
+		model.data = {}
+		model.expected = {}
+		model.random = model[model.distribution](model)
+
+		local file = filePath("test/"..model.distribution.."_data.csv")
+		model.csv = file:read()
+
+		model.timer = Timer{
+			Event{
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: To reduce indentation, just use
```lua
Event{action = function()
    -- ...
end}

File: packages/base/tests/functional/basics/Random.lua [hunk 2]
```diff
@@ -24,6 +24,148 @@
...
+				action = function()
+					table.insert(model.data, model.random:sample())
+					table.insert(model.expected, model.csv[model.timer:getTime()].x)
+				end
+			},
+			Event{
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: Here too:

```lua
Event{start = ..., priority  = ..., action = function()
    -- ...
end}
```

File: packages/base/tests/functional/basics/Random.lua [hunk 3]
```diff
@@ -24,6 +24,148 @@
...
+		local maxArea = 0
+
+		for x = minX, maxX, step do
+			local counter = function(y, v)
+				if math.abs(x - v) < 1 then
+					return y + 1
+				else
+					return y
+				end
+			end
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: Add an empty line after `end`.

File: packages/base/tests/functional/basics/Random.lua [hunk 4]
```diff
@@ -24,6 +24,148 @@
...
+			local y = self:reduce(data, counter)
+			table.insert(xs, x)
+			table.insert(density, y)
+			maxArea = maxArea + x - minX
+		end
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: Same here.

File: packages/base/tests/functional/basics/Random.lua [hunk 5]
```diff
@@ -24,6 +24,148 @@
 --          Rodrigo Reis Pereira
 -------------------------------------------------------------------------------------------
 
+local TestDistributions = Model{
+	finalTime = 1000,
+	distribution = Choice{"exponential", "normal", "lognormal", "beta"},
+	minChart = 0,
+	maxChart = 10,
+	stepChart = 0.5,
+
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: No need for an empty line here to separate the elements of a table.

File: packages/base/tests/functional/basics/Random.lua [hunk 6]
```diff
@@ -24,6 +24,148 @@
...
+				priority = "verylow",
+				action = function()
+					local dfData = model:density(model.data, model.minChart, model.maxChart, model.stepChart)
+					local dfExpected = model:density(model.expected, model.minChart, model.maxChart, model.stepChart)
+					local minimum
+					if #dfData.x < #dfExpected.x then
+						minimum = dfData.x
+					else
+						minimum = dfExpected.x
+					end
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: Add an empty line after an `end` if it is not followed by ` end` or `}`

File: packages/base/tests/functional/basics/Random.lua [hunk 7]
```diff
@@ -24,6 +24,148 @@
...
+					local tme = {}
+					local r = {}
+					local x = {}
+					for i = 1, #minimum do
+						table.insert(tme, dfData.density[i])
+						table.insert(r, dfExpected.density[i])
+						table.insert(x, minimum[i])
+					end
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: here too

File: packages/base/tests/functional/basics/Random.lua [hunk 8]
```diff
@@ -24,6 +24,148 @@
...
+				end
+			}
+		}
+	end,
+	reduce = function(_, data, binaryFunction, initialValue)
+		if not initialValue then initialValue = 0 end
+		local s = initialValue
+		for _, v in pairs(data) do
+			s = binaryFunction(s, v)
+		end
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: add empty line...

File: packages/base/tests/functional/basics/Random.lua [hunk 9]
```diff
@@ -24,6 +24,148 @@
...
+					local tme = {}
+					local r = {}
+					local x = {}
+					for i = 1, #minimum do
+						table.insert(tme, dfData.density[i])
+						table.insert(r, dfExpected.density[i])
+						table.insert(x, minimum[i])
+					end
+					local df = DataFrame{TerraME = tme, R = r, x = x}
+					Chart{target = df, select = {"TerraME", "R"}, xAxis = "x", yLabel = "density", title = model.distribution}
```
Comment (pedro-andrade-inpe) [inline packages/base/tests/functional/basics/Random.lua]: This chart is unnecessary as there is no `UnitTest:assertSnapshot` associated to it.

Comment (wsenafranca) [reply packages/base/tests/functional/basics/Random.lua]: Should I remove this chart or use UnitTest:assertSnapshot?

Comment (pedro-andrade-inpe) [reply packages/base/tests/functional/basics/Random.lua]: Remove the Chart. The assertEquals are enough.

**Pull request merged**